ios/RNSScreenStack.mm

```diff
@@ -57,6 +57,31 @@ - (UIViewController *)childViewControllerForStatusBarHidden
   return [self topViewController];
 }
 
+- (void)viewDidLayoutSubviews
+{
+  if ([self.topViewController isKindOfClass:[RNSScreen class]]) {
+    RNSScreen *screenController = (RNSScreen *)self.topViewController;
+
+    // If RNSScreen includes a navigation controller of type RNScreensNavigationController, it should not calculate
+    // header height, as it could have nested stack.
+    if (!(screenController.childViewControllers.count > 0 && screenController.hasAnyChildNavigators)) {
```

Comment:
Why do we need to check the `count` here? Shouldn't `hasAnyChildNavigators` just return `NO` then?

Reply:
Seems like it should! I'll try tomorrow making an example with nested stack and check if `hasAnyChildNavigators` is sufficient there

Reply:
Can confirm that this part of if statement is unnecessary, I've changed it in https://github.com/software-mansion/react-native-screens/pull/1802/commits/271a4374fa394bf13d7516b348325698d8e86145.